esmvaltool/preprocessor/_reformat.py

```diff
@@ -29,9 +32,12 @@ def _checker(cube):
 
 def fix_file(filename, short_name, project, model, preproc_dir):
     """Fix errors that prevent loading or can not be fixed in the cube."""
+    final_dir = os.path.join(preproc_dir, project, model, short_name)
```

Comment:
Instead of trying to come up with a good path here, I think it would be much better to simply provide the name of the output file (`variable['filename']`) to the function `fix_file` in namelist.py and write the fixed file to that location.

Reply:
We usually have more than on file at this step

Reply:
Ok. I will use msktemp with the variable[filename] as a prefix